database-handler.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+class DatabaseHandler
+{
+    private $table_name;
+
+    public function init()
+    {
+        global $wpdb;
+
+        $this->table_name = $wpdb->prefix . "lightning_publisher_payments";
+
+        $charset_collate = $wpdb->get_charset_collate();
+
+        $sql = "CREATE TABLE $this->table_name (
+        id bigint(20) NOT NULL AUTO_INCREMENT,
+        post_id bigint(20) NOT NULL,
+        payment_hash varchar(256) NOT NULL,
+        payment_request varchar(256) NOT NULL,
+        amount_in_satoshi double(30, 20) DEFAULT 0.0 NOT NULL,
+        exchange_rate double(30, 20) DEFAULT 0.0 NOT NULL,
+        exchange_currency varchar(10) NOT NULL,
+        created_at datetime DEFAULT CURRENT_TIMESTAMP NOT NULL,
+        settled_at datetime DEFAULT CURRENT_TIMESTAMP NOT NULL,
+        state tinytext NOT NULL,
+        PRIMARY KEY  (id)
+        ) $charset_collate;";
+
+        require_once(ABSPATH . 'wp-admin/includes/upgrade.php');
+        dbDelta($sql);
```

Comment:
this thing is also taking care of upgrades/migrations to the table? 
that's why we also do not need any `IF NOT EXISTS`?

Reply:
yes, it's possible to trigger the table creation manually. we can have a setting for that and only create the table if the option is set to true.

line 29 takes care of the database update process. no need to check if the table exists.

Reply:
ok, we can do that in the future. could you create an issue for that that we do not forget it?